### PR TITLE
Chip | FE | Refactor and add size prop

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-1040
http://vjira2:8080/browse/BDS-1540

## Summary

Refactored Chip component to follow current conventions and added spacing prop.

## Details

1. Twig and JS have been completely rewritten to follow current conventions.
2. A new `size` prop is added to the schema.
3. Tag is removed from the schema. It's now only a variable in Twig. The semantic tag of a chip is automatically determined by the `url` prop.
4. Icon slots are added so a chip can accept icon in addition to text.
5. Tests and testing instructions are revised.

## New Features

1. Size options are now available: xsmall, small, medium.
2. Icon support has been added.

## Deprecations

1. `tag` prop is deprecated.

## How to test

Pull down the branch locally and perform the following.

### Check tests

- [x] 1. Given I have pulled down the branch locally
- [x] 2. And I am in the root folder
- [x] 3. When I run the command `npx jest packages/components/bolt-chip/__tests__/chip.js`
- [x] 4. Then I should see all tests indicated as "Pass"

### Check component twig
- [x] 1. Given I am in the `bolt-chip` component folder
- [x] 2. And I go to the `src` folder
- [x] 3. When I open up the `chip.twig` and `chip.js` files
- [x] 4. Then I should see the code is consistent with Bolt's current [standards and conventions](https://github.com/bolt-design-system/bolt/wiki/Code-Standards-and-Conventions)
- [x] 5. And the logics and variables are set correctly

### Check URL prop
- [x] 1. Given I am viewing the URL section of the url variations page
- [x] 2. And I see two chips, one reads "No URL", the other reads "Has URL"
- [x] 3. When I click on the "No URL" chip
- [x] 4. Then nothing should happen
- [x] 5. When I hover over "Has URL" chip
- [x] 6. Then the chip's background color will become darker
- [x] 7. When I click on the "Has URL" chip
- [x] 8. Then the page will be directed to https://pega.com

### Check spacing prop
- [x] 1. Given I am viewing the spacing variations page
- [x] 2. And I see 3 chips that read "xsmall", "small", and "medium"
- [x] 3. When I compare the spacing of each chip
- [x] 4. Then I should see "xsmall" having the smallest spacing and "medium" having the largest spacing

### Check icon usage
- [x] 1. Given I am viewing the icon variations page
- [x] 2. And I see two chips
- [x] 3. When look at the chip that reads "icon before"
- [x] 4. Then I should see an icon before the text
- [x] 5. When look at the chip that reads "icon after"
- [x] 6. Then I should see an icon after the text

### Check no-js version
- [x] 1. Given I am viewing the url variations page
- [x] 2. And I see four chips
- [x] 3. When I turn off javascript in my browser
- [x] 4. Then the styles of the chips should remain the same with the exception of a font change

### CODE REVIEW

File: packages/components/bolt-chip/src/chip.js [hunk 1]
```diff
@@ -1,72 +1,114 @@
...
+      renderedChip = html`<span class="${classes}"
+          >${innerSlots}</span
+        >`;
     }
 
-    return html`
-      ${this.addStyles([styles])} ${renderedChip}
-    `;
+    // [1]
+    // prettier-ignore
```
[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply packages/components/bolt-chip/src/chip.js]: I copy these from the link and trigger component. I believe it is for removing unwanted white space, see the notes (1 and 2) from @danielamorse.

**Comment by mikemai2awesome** [comment]
@adamszalapski 

As discussed with @sghoweri, icons currently do not work without JS.

**Comment by sghoweri** [comment]
> @adamszalapski
> 
> As discussed with @sghoweri, icons currently do not work without JS.

Thankfully I’ve got a few updates in progress that’ll add Bolt Icon support without JS 😉

**Comment by mikemai2awesome** [comment]
@sghoweri 

Good catch, I was on auto drive. This is what code review is for. ;)

Re: default size. Matt requested to make xsmall the default, but I guess they could do that on the drupal side, as for us, maybe we just keep it at small so there's no change.

**Comment by mikemai2awesome** [comment]
@sghoweri I made the changes.

**Comment by sghoweri** [comment]
PR was released with v2.10.0

**Pull request merged**